### PR TITLE
fix monorepo-folders example imports and tsconfigs

### LINKED ISSUE #85: [Bug] monorepo-folders example crashes
### What are you really trying to do?

Trying to run the monorepo-folders example on my machine following the instructions in the README.

### Describe the bug

Got the below error:

```
[frontend] ℹ ｢wds｣: Project is running at http://192.168.0.49/
[frontend] ℹ ｢wds｣: webpack output is served from 
[frontend] ℹ ｢wds｣: Content not from webpack is served from /home/val/Workspace/temporalio/samples-typescript/monorepo-folders/packages/frontend-ui/public
[frontend] ℹ ｢wds｣: 404s will fallback to /
[frontend] Starting the development server...
[frontend] 
[worker] /home/val/Workspace/temporalio/samples-typescript/monorepo-folders/node_modules/ts-node/src/index.ts:750
[worker]     return new TSError(diagnosticText, diagnosticCodes);
[worker]            ^
[worker] TSError: ⨯ Unable to compile TypeScript:
[worker] worker.ts(2,29): error TS2307: Cannot find module 'temporal-workflows/lib/all-activities' or its corresponding type declarations.
[worker] 
[worker]     at createTSError (/home/val/Workspace/temporalio/samples-typescript/monorepo-folders/node_modules/ts-node/src/index.ts:750:12)
[worker]     at reportTSError (/home/val/Workspace/temporalio/samples-typescript/monorepo-folders/node_modules/ts-node/src/index.ts:754:19)
[worker]     at getOutput (/home/val/Workspace/temporalio/samples-typescript/monorepo-folders/node_modules/ts-node/src/index.ts:941:36)
[worker]     at Object.compile (/home/val/Workspace/temporalio/samples-typescript/monorepo-folders/node_modules/ts-node/src/index.ts:1243:30)
[worker]     at Module.m._compile (/home/val/Workspace/temporalio/samples-typescript/monorepo-folders/node_modules/ts-node/src/index.ts:1370:30)
[worker]     at Module._extensions..js (internal/modules/cjs/loader.js:1114:10)
[worker]     at Object.require.extensions.<computed> [as .ts] (/home/val/Workspace/temporalio/samples-typescript/monorepo-folders/node_modules/ts-node/src/index.ts:1374:12)
[worker]     at Module.load (internal/modules/cjs/loader.js:950:32)
[worker]     at Function.Module._load (internal/modules/cjs/loader.js:790:12)
[worker]     at Function.executeUserEntryPoint [as runMain] (internal/modules/run_main.js:76:12) {
[worker]   diagnosticText: "worker.ts(2,29): error TS2307: Cannot find module 'temporal-workflows/lib/all-activities' or its corresponding type declarations.\n",
[worker]   diagnosticCodes: [ 2307 ]
[worker] }
[worker] [nodemon] app crashed - waiting for file changes before starting...
[frontend] Compiled successfully!
[frontend] 
[frontend] You can now view frontend-ui in the browser.
[frontend] 
[frontend]   Local:            http://localhost:3000
[frontend]   On Your Network:  http://192.168.0.49:3000
[frontend] 
[frontend] Note that the development build is not optimized.
[frontend] To create a production build, use npm run build.
[frontend] 
[api-server] TypeError: Cannot read property 'QUERY_REJECT_CONDITION_UNSPECIFIED' of undefined
[api-server]     at defaultWorkflowClientOptions (/home/val/Workspace/temporalio/samples-typescript/monorepo-folders/node_modules/@temporalio/client/src/workflow-client.ts:160:70)
[api-server]     at new WorkflowClient (/home/val/Workspace/temporalio/samples-typescript/monorepo-folders/node_modules/@temporalio/client/src/workflow-client.ts:207:25)
[api-server]     at Object.runWorkflow (/home/val/Workspace/temporalio/samples-typescript/monorepo-folders/packages/backend-apis/temporal-client.ts:9:18)
[api-server]     at /home/val/Workspace/temporalio/samples-typescript/monorepo-folders/packages/backend-apis/server.ts:14:41
[api-server]     at Layer.handle [as handle_request] (/home/val/Workspace/temporalio/samples-typescript/monorepo-folders/node_modules/express/lib/router/layer.js:95:5)
[api-server]     at next (/home/val/Workspace/temporalio/samples-typescript/monorepo-folders/node_modules/express/lib/router/route.js:137:13)
[api-server]     at Route.dispatch (/home/val/Workspace/temporalio/samples-typescript/monorepo-folders/node_modules/express/lib/router/route.js:112:3)
[api-server]     at Layer.handle [as handle_request] (/home/val/Workspace/temporalio/samples-typescript/monorepo-folders/node_modules/express/lib/router/layer.js:95:5)
[api-server]     at /home/val/Workspace/temporalio/samples-typescript/monorepo-folders/node_modules/express/lib/router/index.js:281:22
[api-server]     at Function.process_params (/home/val/Workspace/temporalio/samples-typescript/monorepo-folders/node_modules/express/lib/router/index.js:335:12)
[api-server] [nodemon] app crashed - waiting for file changes before starting...
[frontend] Proxy error: Could not proxy request /api/workflow from localhost:3000 to http://localhost:4000.
[frontend] See https://nodejs.org/api/errors.html#errors_common_system_errors for more information (ECONNRESET).
[frontend] 

```

### Minimal Reproduction

monorepo-folders

### Environment/Versions

yarn 1.22.17
ubuntu 20.04

- OS and processor: [e.g. M1 Mac, x86 Windows, Linux] ubuntu 20.04
- Temporal Version: [e.g. 1.14.0?] and/or SDK version: SDK 0.16.4, temporalio/docker-compose v1.13.1
- Are you using Docker or Kubernetes or building Temporal from source?

### PR DESCRIPTION
**Pull request by vkarpov15**
## What was changed

There were several issues in the configs of the monorepo-folders example:

1. As @lorensr mentioned in https://github.com/temporalio/samples-typescript/issues/85#issuecomment-1009654002, incorrect `compilerOptions.rootDir` in temporal-workflows
2. Inconsistent SDK version in temporal-worker/package.json caused the "Cannot read property 'QUERY_REJECT_CONDITION_UNSPECIFIED' of undefined" error
3. Incorrect `rootDir` in top-level `package.json`
4. Looks like an incorrect import path in `temporal-worker/worker.ts`

## Why?

monorepo-folders example failed to run, see #85

## Checklist

1. Closes #85

2. How was this tested:

Ran the example locally and looks like it works: 

image

3. Any docs updates needed?

No

### CODE REVIEW

File: monorepo-folders/packages/backend-apis/temporal-client.ts [hunk 1]
```diff
@@ -13,13 +13,13 @@ export async function runWorkflow() {
   // Invoke the `WorkflowA` Workflow, only resolved when the workflow completes
   const result = await client.execute(WorkflowA, {
     taskQueue: 'tutorial',
-    workflowId: 'workflow-a',
+    workflowId: 'workflow-a-' + Date.now(),
```
Comment (bergundy) [inline monorepo-folders/packages/backend-apis/temporal-client.ts:16]: Why did we add the date here?

File: monorepo-folders/tsconfig.json [hunk 2]
```diff
@@ -5,7 +5,7 @@
     "declaration": true,
     "declarationMap": true,
     "sourceMap": true,
-    "rootDir": "./src",
+    "rootDir": "./packages",
```
[the remark this answers is not in the document]
Comment (vkarpov15) [reply monorepo-folders/tsconfig.json:8]: Nope, still compiles without it. Removed

File: monorepo-folders/packages/backend-apis/temporal-client.ts (hunk 1 above)
Comment (vkarpov15) [reply monorepo-folders/packages/backend-apis/temporal-client.ts:16]: Worth @sw-yx 's thoughts here. My issue with this as implemented was that I couldn't run the workflow twice from the UI, the 2nd time I'd get "ALREADY_EXISTS: Workflow execution is already running. WorkflowId: workflow-a". Which is a bit of friction.

What do you think?

**Comment by lorensr** [comment]
`backend-apis` start script should be:

```
    "start": "nodemon server.ts --watch . --watch ../temporal-workflows"
```

> Don't use unix globbing to pass multiple directories, e.g --watch ./lib/*, it won't work. You need a --watch flag per directory watched.

https://github.com/remy/nodemon#monitoring-multiple-directories

File: monorepo-folders/packages/backend-apis/temporal-client.ts (hunk 1 above)
Comment (bergundy) [reply monorepo-folders/packages/backend-apis/temporal-client.ts:16]: It shouldn't return ALREADY_EXISTS if the workflow finishes.
I'd at least add a comment that users should provide a meaningful business ID.

**Review by bergundy — APPROVED** [review]
Nothing blocking here

File: monorepo-folders/packages/backend-apis/temporal-client.ts (hunk 1 above)
Comment (sw-yx) [reply monorepo-folders/packages/backend-apis/temporal-client.ts:16]:
```suggestion
    workflowId: 'workflow-a-' + Date.now(), // TODO: remember to replace this with a meaningful business ID
```

File: monorepo-folders/packages/backend-apis/temporal-client.ts [hunk 3]
```diff
@@ -13,13 +13,13 @@ export async function runWorkflow() {
...
     args: ['Temporal'], // type inference works! args: [name: string]
   });
   // Starts the `WorkflowB` Workflow, don't wait for it to complete
   await client.start(WorkflowB, {
     taskQueue: 'tutorial',
-    workflowId: 'workflow-b',
+    workflowId: 'workflow-b-' + Date.now(),
```
[the remark this answers is not in the document]
Comment (sw-yx) [reply monorepo-folders/packages/backend-apis/temporal-client.ts:22]:
```suggestion
    workflowId: 'workflow-b-' + Date.now(), // TODO: remember to replace this with a meaningful business ID
```

**Review by sw-yx — APPROVED** [review]

**Pull request merged**